ward/cparser.lua

```diff
@@ -1088,8 +1088,16 @@ local grammar = pattern {
     function(str, pos, start, decls_and_stmts, finish)
       return pos, { build_compound_stmt, start, finish, decls_and_stmts }
     end),
+  -- Warning: the following rule actually ignores the initialization
+  -- part of a for statement if it includes a type declaration. The
+  -- implied assumption is that for variables should not actually be of
+  -- type Obj or Bag, as we may not be able to properly generate guards
+  -- for such variables in the for header, anyway. Ideally, we should
+  -- issue a warning if we see one.
+  for_init_decl = action(rule "type_prefix" * sp *
```

Comment:
It would be nice if the comment also mentioned what the code is for -- namely support for C99 "for" loops with variable declarations in the "init" section. Perhaps: "The following hack allows ward to parse for loops which declare variables in their init section, like e.g. `for (int i = 0; i < n; i++) { ... }`